meilisearch/tests/index/test_index_search_meilisearch.py

```diff
@@ -1,40 +1,54 @@
-import time
 import json
 import meilisearch
 from meilisearch.tests import BASE_URL, MASTER_KEY
 
 class TestSearch:
+
+    """ TESTS: search route """
+
     client = meilisearch.Client(BASE_URL, MASTER_KEY)
     index = None
+    dataset_file = None
+    dataset_json = None
 
     def setup_class(self):
         self.index = self.client.create_index(uid='indexUID')
-        json_file = open('./datasets/small_movies.json')
-        data = json.load(json_file)
-        self.index.add_documents(data, primary_key='id')
-        time.sleep(1)
+        self.dataset_file = open("./datasets/small_movies.json", "r")
+        self.dataset_json = json.loads(self.dataset_file.read())
+        self.dataset_file.close()
+        response = self.index.add_documents(self.dataset_json, primary_key='id')
+        self.index.wait_for_pending_update(response['updateId'])
 
     def teardown_class(self):
         self.index.delete()
 
+    def test_basic_search_empty_query(self):
+        """Tests search with an empty query"""
+        response = self.index.search('')
+        assert len(response['hits']) == 0
+
     def test_basic_search(self):
+        """Tests search with an simple query"""
         response = self.index.search('How to Train Your Dragon')
         assert isinstance(response, object)
         assert response['hits'][0]['id'] == '166428'
 
```

Comment:
There should be three additionals tests that tests the following values during a search.

On the query parameters that accepts either an asterix, a string, or an array or string

- One tests that tries query parameters with a "*" value
- One tests that tries query parameters with a simple string value
- One tests that tries query parameters with an array of string as value

Better explaination was made in the PHP sdk: meilisearch/meilisearch-php#19 (comment)

This explanation does not add the tests on a simple string, as it is potentially breaking I would prefer that it is also tested.
The reason I mentioned those tests is because of the hight probability that the codes act in a certain way depending on the type of data it expects. For example the JS library crashed on a simple string before those tests were done.

Reply:
The first two tests were added. The third one actually uncovered a bug, detailed in #85. Test was removed and will be integrated when that issue is solved.